includes/discovery/mempools/edgecos.inc.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/*
+ * LibreNMS 
+ *
+ * Copyright (c) 2017 Martin Zatloukal <slezi2@pvfree.net>
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+if ($device['os'] == 'edgecos') {
+    d_echo('EdgeCore Memory:');
+
+    $temp_total = snmp_get($device, '.1.3.6.1.4.1.259.10.1.24.1.39.3.1.0', '-Ovqn');//ECS4510
+    if (!empty($temp_total)) {
+         $total = snmp_get($device, '.1.3.6.1.4.1.259.10.1.24.1.39.3.1.0', '-Ovqn');
```

Comment:
No point in running this snmp_get, you already have this data.

You then should check if `$total and $avail` are set for the rest of these so you don't continuously send snmp gets if you match early on.

Also, can you not do anything with sysObjectID to work out what the OIDs are?

Reply:
The use of sysObjectID is not possible because it is not known yet in discovery

```
#### Load disco module processors ####
Array
(
    [device_id] => 39
    [hostname] => 192.168.224.254
    [sysName] => 192.168.224.254
    [ip] =>.
    [community] => public
    [authlevel] =>.
    [authname] =>.
    [authpass] =>.
    [authalgo] =>.
    [cryptopass] =>.
    [cryptoalgo] =>.
    [snmpver] => v2c
    [port] => 161
    [transport] => udp
    [timeout] =>.
    [retries] =>.
    [snmp_disable] => 0
    [bgpLocalAs] =>.
    [sysObjectID] =>.
    [sysDescr] =>.
    [sysContact] =>.
    [version] =>.
    [hardware] =>.
    [features] =>.
    [location] =>.
    [os] => edgecos
    [status] => 1
    [status_reason] =>.
    [ignore] => 0
    [disabled] => 0
    [uptime] =>.
    [agent_uptime] => 0
    [last_polled] =>.
    [last_poll_attempted] =>.
    [last_polled_timetaken] =>.
    [last_discovered_timetaken] =>.
    [last_discovered] =>.
    [last_ping] =>.
    [last_ping_timetaken] =>.
    [purpose] =>.
    [type] => network
    [serial] =>.
    [icon] => edge-core.png
    [poller_group] => 0
    [override_sysLocation] => 0
    [notes] =>.
    [port_association_mode] => 1
    [attribs] => Array
        (
        )

    [snmp_max_repeaters] =>.
)
```

Reply:
It should be and if it isn't / doesn't we should make sure it is.

You could however just do the sysObjectID lookup once anyway to still save all of this but I favor fixing disco.

Reply:
Should already be available in `$device['sysObjectID']`

Reply:
When adding a new device, $ device ['sysObjectID'] is only available after the first polling